bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -211,50 +211,54 @@ private void refresh() {
-            updateQtyState(CHANNEL_SENEC_CURRENT_PACK4, response.battery.current[3], 2, Units.AMPERE);
-            updateQtyState(CHANNEL_SENEC_VOLTAGE_PACK1, response.battery.voltage[0], 2, Units.VOLT);
-            updateQtyState(CHANNEL_SENEC_VOLTAGE_PACK2, response.battery.voltage[1], 2, Units.VOLT);
-            updateQtyState(CHANNEL_SENEC_VOLTAGE_PACK3, response.battery.voltage[2], 2, Units.VOLT);
-            updateQtyState(CHANNEL_SENEC_VOLTAGE_PACK4, response.battery.voltage[3], 2, Units.VOLT);
-            updateQtyState(CHANNEL_SENEC_MAX_CELL_VOLTAGE_PACK1, response.battery.maxCellVoltage[0], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MAX_CELL_VOLTAGE_PACK2, response.battery.maxCellVoltage[1], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MAX_CELL_VOLTAGE_PACK3, response.battery.maxCellVoltage[2], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MAX_CELL_VOLTAGE_PACK4, response.battery.maxCellVoltage[3], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MIN_CELL_VOLTAGE_PACK1, response.battery.minCellVoltage[0], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MIN_CELL_VOLTAGE_PACK2, response.battery.minCellVoltage[1], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MIN_CELL_VOLTAGE_PACK3, response.battery.minCellVoltage[2], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
-            updateQtyState(CHANNEL_SENEC_MIN_CELL_VOLTAGE_PACK4, response.battery.minCellVoltage[3], 3, Units.VOLT,
-                    DIVISOR_MILLI_TO_ISO);
+            // We are checking only one field here. It is assumed that other fields, like dischargedEnergy, cycles, etc.
+            // are available then, too, since these are typical for newer Senec systems with LiPo batteries.
+            if (response.battery.chargedEnergy != null) {
+                updateQtyState(CHANNEL_SENEC_CHARGED_ENERGY_PACK1, response.battery.chargedEnergy[0], 2,
+                        Units.KILOWATT_HOUR, DIVISOR_MILLI_TO_KILO);
+                updateQtyState(CHANNEL_SENEC_CHARGED_ENERGY_PACK2, response.battery.chargedEnergy[1], 2,
+                        Units.KILOWATT_HOUR, DIVISOR_MILLI_TO_KILO);
+                updateQtyState(CHANNEL_SENEC_CHARGED_ENERGY_PACK3, response.battery.chargedEnergy[2], 2,
+                        Units.KILOWATT_HOUR, DIVISOR_MILLI_TO_KILO);
+                updateQtyState(CHANNEL_SENEC_CHARGED_ENERGY_PACK4, response.battery.chargedEnergy[3], 2,
+                        Units.KILOWATT_HOUR, DIVISOR_MILLI_TO_KILO);
+                updateQtyState(CHANNEL_SENEC_DISCHARGED_ENERGY_PACK1, response.battery.dischargedEnergy[0], 2,
```

Comment:
Hey @thopiekar , how can you ensure that `dischargedEnergy` is not null? Just because `chargedEnergy` is not null? I have not digged into this. @eguib did a nice proposal here: https://github.com/openhab/openhab-addons/issues/11606#issuecomment-1078840338 , could you take this over?

Reply:
Yes, I can pick the patch and mention him here as an author, if you and @eguib like :)